### ISSUE TITLE
Point at null island leads to "Error calculating min/max zoom: Bounds invalid"

### ISSUE THREAD

**Issue by springmeyer** [mapbox/mapnik-omnivore#88]
The below json will trigger an error from omnivore like `Error calculating min/max zoom: Bounds invalid`. This seems odd since it is a valid point on null island. Though i could see the argument that this type of data might not be common. @GretaCB is there a reason you see for this error being correct? Unless you think otherwise my sense is that this should not error out and should instead result in `[0,0,0,0]` for the bounds.

Btw, I also noticed that an empty shapefile (with zero features) also leads to this error. And it trickles up to the user in a modal window in Studio (at least with the omnivore version in studio). To test you can fetch the `3d_point_empty.shp from: https://github.com/mapnik/mapnik/blob/master/tests/data/shp/

```json
{
  "type": "FeatureCollection",
  "features": [
    {
      "type": "Feature",
      "properties": { },
      "geometry": {
        "type": "Point",
        "coordinates": [ 0, 0 ]
      }
    }
  ]
}
```

**Comment by rclark** [comment]
This is actually popping out from where omnivore is [attempting to determine the min/max zoom for the file](https://github.com/mapbox/mapnik-omnivore/blob/fac6d442d8a13214ac3b2c4f0a08cfc4a25e5cee/lib/datasource-processor.js#L172-L181). Its basically stretching the file's size across the number of tiles that will be rendered at successive zoom ranges, and crossing a threshold marks the bounds.

The real question might be: given this feature, what is the desired z-extent? Could we pad all extents by a wee bit which would throw this file into 4 tiles at every zoom range?

**Comment by GretaCB** [comment]
Taking a gander at this, post-mapnik update in unpacker. Thinking of adding special logic for `0,0` specifically and default to 1 tile. @rclark would padding be better? 

cc @lyzidiamond

**Comment by rclark** [comment]
I think padding would be better. Strictly speaking, this point _is not in any tiles_ -- it is right on the corner of 4 tiles from max zoom to z1, and then it **is** in the z0 tile. 

By padding the point's bbox a bit, you simulate the fact that mapnik's tile buffers would incorporate the point into each of the 4 tiles that it lies on the edge of.

**Comment by BergWerkGIS** [comment]
This also surfaced in Mapbox Studio: https://github.com/mapbox/mapbox-studio/issues/1447

Source in `mapnik-omnivore`, where this is coming from:
https://github.com/mapbox/mapnik-omnivore/blob/ce7f47af9135acdc144991d5ba27469d76e5649f/lib/utils.js#L22-L29

With the data from the Mapbox Studio issues, it works fine from zMax to z8, at z7, this algo fails:

![image](https://cloud.githubusercontent.com/assets/4549888/9817030/d32f0676-58a0-11e5-8ec9-5b092bb72a40.png)

**Comment by BergWerkGIS** [comment]
This variable values in the for loop:

```
z:22    x:94    y:94    minX:2097105    minY:2097105    maxX:2097198    maxY:2097198    tiles:8836
z:21    x:48    y:48    minX:1048552    minY:1048552    maxX:1048599    maxY:1048599    tiles:2304
z:20    x:24    y:24    minX:524276     minY:524276     maxX:524299     maxY:524299     tiles:576
z:19    x:12    y:12    minX:262138     minY:262138     maxX:262149     maxY:262149     tiles:144
z:18    x:6     y:6     minX:131069     minY:131069     maxX:131074     maxY:131074     tiles:36
z:17    x:4     y:4     minX:65534      minY:65534      maxX:65537      maxY:65537      tiles:16
z:16    x:2     y:2     minX:32767      minY:32767      maxX:32768      maxY:32768      tiles:4
z:15    x:2     y:2     minX:16383      minY:16383      maxX:16384      maxY:16384      tiles:4
z:14    x:2     y:2     minX:8191       minY:8191       maxX:8192       maxY:8192       tiles:4
z:13    x:2     y:2     minX:4095       minY:4095       maxX:4096       maxY:4096       tiles:4
z:12    x:2     y:2     minX:2047       minY:2047       maxX:2048       maxY:2048       tiles:4
z:11    x:2     y:2     minX:1023       minY:1023       maxX:1024       maxY:1024       tiles:4
z:10    x:2     y:2     minX:511        minY:511        maxX:512        maxY:512        tiles:4
z:9     x:2     y:2     minX:255        minY:255        maxX:256        maxY:256        tiles:4
z:8     x:2     y:2     minX:127        minY:127        maxX:128        maxY:128        tiles:4
z:7     x:0     y:0     minX:64 minY:64 maxX:63 maxY:63 tiles:0
<Fri, 11 Sep 2015 14:54:59 GMT> error 500 { [Error: Error calculating min/max zoom: Bounds invalid]
  code: 'EINVALID',
  method: 'GET',
  url: '/metadata?file=c:/mb/_TEMP\\extent.json' }
```

**Comment by GretaCB** [comment]
Relevant to https://github.com/mapbox/node-sphericalmercator/pull/13 ?

**Comment by lyzidiamond** [comment]
A user seeing this on Twitter: https://twitter.com/mgiraldo/status/648625171683770369

**Comment by springmeyer** [comment]
Confirmed that this was fixed by sphericalmercator@1.0.4. Added test coverage for this in 841391c. Will be live once applications upgrade to the upcoming mapnik-om@6.3.2: https://github.com/mapbox/mapnik-omnivore/milestones/6.3.2
